WordPress/Classes/Login/WordPressDotComAuthenticator.swift

```diff
@@ -9,46 +9,176 @@ import Alamofire
 ///
 /// API doc: https://developer.wordpress.com/docs/oauth2/
 struct WordPressDotComAuthenticator {
-    enum Error: Swift.Error {
+    enum SignInContext {
+        // Sign in to the app with a WordPress.com account.
+        // Using this context results in automatically reloading the app to display an site in the account.
+        case `default`
+        // Sign in to an existing account.
+        // This is typically used when the app needs to get a new OAuth token because the current one becomes invalid.
+        case reAuthentication(TaggedManagedObjectID<WPAccount>)
+        // Connect a site to Jetpack or sign in to an already connected site.
+        case jetpackSite(accountEmail: String?)
+
+        func accountEmail(in context: NSManagedObjectContext) -> String? {
+            switch self {
+            case .default:
+                return nil
+            case let .reAuthentication(accountID):
+                return (try? context.existingObject(with: accountID))?.email
+            case let .jetpackSite(email):
+                return email
+            }
+        }
+    }
+
+    enum AuthenticationError: Error {
```

Comment:
(nit) There are scenarios presented in these cases that the user of this class probably shouldn't not about as there s no good way to resolve them or present to the user.

Reply:
Do you mean cases like `parsing` and `invalidCallbackURL` do not need to be in this enum? Like putting them into an umbrella `unknown` or `other` cases?